### PR TITLE
Fix: Doctor search and filter screens styling 

### LINKED ISSUE #56: Fix:Doctors search and filter styling
This issue fixes stying of the modal so that it is responsive  in all devices and also changing where the filter modal will be opened from which in this case will be the filter Icon

### PR DESCRIPTION
**Pull request by Latiah**
What does this PR do?

This Pr fixes some  designs that were not responsive on all devices and also add the modal to open when someone clicks on the filter icon on the search input

**Tasks to be completed**

- [x] Fix the modal component to be responsive in all devices as in some it was somehow not fitting the screen as intended
- [x] change the position where someone has to click to open the filter  modal 

How to test this manually 

click on the fix filter modal on the first screen that appears and you will be directed to the doctors found and click on the filter icon to open the modal

### CODE REVIEW

File: components/searchinput2.tsx
Comment (fredshema) [inline components/searchinput2.tsx:1]: I would suggest using native wind for styles instead of using stylesheets.

Comment (Latiah) [reply components/searchinput2.tsx:1]: Hey @fredshema  I just changed some styles to native wind but there are some few I did not change .

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "expo-updates": "~0.25.14",
     "jest": "^29.7.0",
     "nativewind": "^2.0.11",
+    "npm": "^10.8.0",
```
Comment (fredshema) [inline package.json]: Is this package needed? if not you can remove it

Comment (Latiah) [reply package.json]: My app was not starting and they suggested to  add It  that's how it came in .but let me check if it will work if I remove it

Comment (Latiah) [reply package.json]: I just removed the package

**Review by fredshema — APPROVED** [review]

**Comment by Emmyfrank** [comment]
1: work on the alignment of top -left back button, you can check on the following screen


PHOTO-2024-05-21-19-15-17 2

2:you can also fix the alignment of reset and Appy buttons on the bottom of the screen, chech the following screen.

PHOTO-2024-05-21-19-15-17

**Comment by Emmyfrank** [comment]
you can work on those comments and solve your branch's conflict

**Comment by Latiah** [comment]
I do not know why that is happening those screens look fine on my end but I am going to see what I can change

**Comment by Latiah** [comment]
Screenshot_1716312766
Screenshot_1716312761

**Review by 911samuel — APPROVED** [review]

**Review by justinmihigo — APPROVED** [review]
looks well on my end, great job

**Review by chrysloi — COMMENTED** [review]
Please do consider adding an option of closing the modal when tapped outside of it!

**Comment by Latiah** [comment]
> Please do consider adding an option of closing the modal when tapped outside of it!

Thank you .I just added it too

**Comment by chrysloi** [comment]
> > Please do consider adding an option of closing the modal when tapped outside of it!
> 
> Thank you .I just added it too

good Work!

**Review by chrysloi — APPROVED** [review]

**Pull request merged**